### PR TITLE
[SW-309] Allow register of names with all characters allowed in names spec

### PR DESCRIPTION
**Pull request by pstachel**

### CODE REVIEW

File: src/popup/components/TransactionInfo.vue [hunk 1]
```diff
@@ -14,14 +14,18 @@
     </div>
 
     <div class="parties">
-      <Avatar v-bind="sender" />
+      <Avatar
+        v-bind="sender"
+        :name="punycodeToName(sender.name)"
```
Comment (konradodo) [inline src/popup/components/TransactionInfo.vue:19]: punycodeToName(sender.name) -> move to computed, you run this function twice

File: src/popup/components/TransferReview.vue [hunk 2]
```diff
@@ -130,7 +131,8 @@ export default {
       'tippingV2',
     ]),
     isRecipientName() {
-      return this?.recipientAddress && checkAensName(this.recipientAddress);
+      return this.transferData?.address
```
Comment (peronczyk) [inline src/popup/components/TransferReview.vue:134]: I suggest to make the multiline conditions in a way that every item is in separate line.
This way:
```
return (
  condition
  && condition
)
```

File: src/popup/utils/names.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { toASCII, toUnicode } from 'tr46';
+
+const options = {
```
Comment (peronczyk) [inline src/popup/utils/names.ts:3]: I suggest to rename this to `PUNYCODE_OPTIONS`.

File: src/popup/components/TransactionInfo.vue [hunk 4]
```diff
@@ -14,14 +14,18 @@
...
+      />
       <div class="mid">
         <TriangleRight class="triangle" />
         <div class="line" />
       </div>
       <Avatar
         v-if="recipient.address"
         v-bind="recipient"
+        :name="punycodeToName(recipient.name)"
```
Comment (davidyuk) [inline src/popup/components/TransactionInfo.vue:28]: This is something that should be on the sdk side, it is the right way to encode names in not english?

Comment (pstachel) [reply src/popup/components/TransactionInfo.vue:28]: Yes, it's encoded in accordance with the [documentation](https://docs.aeternity.com/protocol/AENS.html#name)

**Pull request closed without merging**